validator/project_validator.go

```diff
@@ -548,17 +548,21 @@ func checkProjectFields(project *model.Project) ValidationErrors {
 
 // Ensures that:
 // 1. a referenced task within a buildvariant task object exists in
-// the set of project tasks
+// the set of project tasks and is not referenced in the task group of the buildvariant
```

Comment:
Nit: should end sentences with periods and start with capitalized sentences (this was wrong before your change but it's hard to read)